storage/replica_command.go

```diff
@@ -853,39 +872,31 @@ func (r *Replica) RangeLookup(
 			if err != nil {
 				return reply, nil, err
 			}
-			// If this is a descriptor we're allowed to return,
-			// do just that and call it a day.
+			// If this is a descriptor we're allowed to return, add that
+			// to the lookup response slice and stop searching in intents.
 			if rd != nil {
-				kvs = []roachpb.KeyValue{{Key: intent.Key, Value: *val}}
-				rds = []roachpb.RangeDescriptor{*rd}
+				reply.Ranges = append(reply.Ranges, *rd)
 				break
```

Comment:
I'll give this a thorough review tomorrow.

On Wed, Mar 30, 2016 at 3:17 PM Nathan VanBenschoten <
notifications@github.com> wrote:

> In storage/replica_command.go
> <https://github.com/cockroachdb/cockroach/pull/5627#discussion_r57946765>:
>
> >  			if rd != nil {
> > -				kvs = []roachpb.KeyValue{{Key: intent.Key, Value: *val}}
> > -				rds = []roachpb.RangeDescriptor{*rd}
> > +				reply.Ranges = append(reply.Ranges, *rd)
> >  				break
>
> Do we want to break here, or return the descriptors from all intents?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/5627/files/86e773a9551ffe6172917ee350b16f7182b54deb#r57946765>
>